models/dataset.go

```diff
@@ -454,24 +454,47 @@ func (ed *EditionUpdate) PublishLinks(ctx context.Context, host string, versionL
 
 // ValidateDataset checks the dataset has invalid fields
 func ValidateDataset(ctx context.Context, dataset *Dataset) error {
+
 	var invalidFields []string
 	if dataset.URI != "" {
 		dataset.URI = strings.TrimSpace(dataset.URI)
-		datasetURI, err := url.Parse(dataset.URI)
+		_, err := url.Parse(dataset.URI)
+		if err != nil {
+			invalidFields = append(invalidFields, "URI")
+			log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
+		}
+	}
+
+	// checks relevant fields within Dataset to ensure all HRef fields are valid
+	for _, list := range [][]GeneralDetails{dataset.Publications, dataset.RelatedDatasets, dataset.Methodologies} {
+		err := trimHref(ctx, list)
 		if err != nil {
 			invalidFields = append(invalidFields, "URI")
 			log.Event(ctx, "error parsing URI", log.ERROR, log.Error(err))
-		} else {
-			if !strings.Contains(datasetURI.Path, "/datasets") || !strings.Contains(datasetURI.Path, dataset.ID) {
-				fmt.Println("hello")
-				invalidFields = append(invalidFields, "URI")
-			}
 		}
 	}
+
 	if invalidFields != nil {
 		return fmt.Errorf("invalid fields: %v", invalidFields)
 	}
 	return nil
+
+}
+
+func trimHref(ctx context.Context, generalDetails []GeneralDetails) error {
+	var invalidFields []string
+	for i := range generalDetails {
+		details := &generalDetails[i]
+		details.HRef = strings.TrimSpace(details.HRef)
```

Comment:
We are updating the Href of a details item that has been brought in to this function, but we are receiving the list as a non-pointer type. Surely the function will be simply updating a copy rather than the original? If the intention is to validate rather than update (as the name implies) then this brings in the question why the `ValidateDataset()` function is changing the href (on line 460) as it's receiving the dataset struct as a pointer.
I believe either this function or the other might need to be updated.